src/main/java/com/example/bookstore/validation/FieldsMatch.java

```diff
@@ -0,0 +1,23 @@
+package com.example.bookstore.validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsMatch {
+    String field();
+
+    String fieldMatch();
```

Comment:
would it be better to:
```suggestion
    String[] fields();
```

and then:
```
@FieldMatch(fields = {"password", "repeatPassword"})
```